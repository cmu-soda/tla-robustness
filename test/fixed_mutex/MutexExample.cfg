SPECIFICATION Spec
INVARIANT TypeOK
INVARIANT MutexProperty
