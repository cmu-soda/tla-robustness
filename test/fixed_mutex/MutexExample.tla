------------------------------- MODULE MutexExample -------------------------------

VARIABLES state, pc1, pc2

vars == <<state, pc1, pc2>>
mutexVars == <<state>>
procVars == <<pc1, pc2>>

Mutex == INSTANCE Mutex
            WITH state <- state

Proc == INSTANCE Proc
            WITH pc1 <- pc1,
                 pc2 <- pc2

Init == Mutex!Init /\ Proc!Init

Down == Mutex!Down /\ Proc!Down

Up == Mutex!Up /\ Proc!Up

Next ==
    \/ Up
    \/ Down

Spec == Init /\ [][Next]_vars

TypeOK == Mutex!TypeOK /\ Proc!TypeOK

MutexProperty == pc1 # "cs" \/ pc2 # "cs"

=============================================================================
