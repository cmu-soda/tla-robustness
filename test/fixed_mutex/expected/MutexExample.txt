TLA+ Module: MutexExample
Diff rep grouped by action:

Down:
Safety boundary representation: out/MutexExample_Down_diff_rep.txt
Safety boundary rep formula (with sort definitions):
Sort1 == {"cs","safe"}
/\ state = "down"
/\ \A S \in Sort1 : (pc2 # S \/ pc1 # S)
